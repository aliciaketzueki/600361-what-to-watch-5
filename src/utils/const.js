export const MORE_LIKE_NUM = 4;
export const MAX_ACTORS_NUM = 4;
export const SECOND = 1000;

export const tabs = [{
  name: `Overview`,
  id: 0,
}, {
  name: `Details`,
  id: 1,
}, {
  name: `Reviews`,
  id: 2
}];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
