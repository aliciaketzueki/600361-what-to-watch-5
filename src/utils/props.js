import PropTypes from "prop-types";

export const validFilm = PropTypes.oneOfType([
  PropTypes.oneOf([null]).isRequired,
  PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired
  }).isRequired
]);

export const validBool = PropTypes.bool.isRequired;
export const validShape = PropTypes.shape().isRequired;
export const validArrayOfShape = PropTypes.arrayOf(PropTypes.shape()).isRequired;
export const validNum = PropTypes.number.isRequired;
export const validArrayOfString = PropTypes.arrayOf(PropTypes.string).isRequired;
export const validFunc = PropTypes.func.isRequired;
export const validString = PropTypes.string.isRequired;
export const validOneOfType = PropTypes.oneOfType([
  PropTypes.func,
  PropTypes.shape({current: PropTypes.any})
]);
