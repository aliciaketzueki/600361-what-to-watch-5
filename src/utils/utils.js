import genres from "../mocks/genres";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getRandom = (min, max) => {
  return Math.floor(Math.random() * (max - min) + min);
};

export const createFilmCard = (arr) => {
  const filmArr = [];

  [].forEach.call(arr, (name) => {
    const film = {};
    film.year = getRandom(1990, 2020);
    film.videoSrc = `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`;
    film.id = getRandom(1000, 9999);
    film.name = name;
    film.poster = film.name.split(` `).join(`-`).split(/[^a-zA-Z0-9_\-]/).join(``).toLowerCase() + `.jpg`;
    film.genreId = getRandom(0, 9);
    film.genre = genres[film.genreId].name;
    filmArr.push(film);
  });

  return filmArr;
};
