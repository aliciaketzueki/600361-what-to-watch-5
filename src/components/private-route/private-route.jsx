import React from "react";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AuthorizationStatus, AppRoute} from "../../utils/const";
import {getAuthorizationStatus} from "../../store/selectors";
import {validString, validFunc} from "../../utils/props";

const PrivateRoute = (props) => {
  const {render, path, authorizationStatus} = props;

  return (
    <Route
      exact
      path={path}
      render={(routeProps) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(routeProps)
            : <Redirect to={AppRoute.LOGIN} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: validString,
  path: validString,
  render: validFunc,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
