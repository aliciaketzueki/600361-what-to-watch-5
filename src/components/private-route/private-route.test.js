import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route";
import {AppRoute} from "../../utils/const";
import {BrowserRouter as Router} from "react-router-dom";
import {AuthorizationStatus} from "../../utils/const";

const MockComponent = () => <div />;

describe(`Render PrivateRoute`, () => {
  test.each([
    [`with`, AuthorizationStatus.AUTH],
    [`without`, AuthorizationStatus.NO_AUTH],
  ])(`%s auth`, (_expected, login) => {
    const tree = renderer
    .create(
        <Router>
          <PrivateRoute
            render={MockComponent}
            path={AppRoute.ROOT}
            authorizationStatus={login}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
  });
});
