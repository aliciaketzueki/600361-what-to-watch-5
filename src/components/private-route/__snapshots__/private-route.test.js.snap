// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute with auth 1`] = `<div />`;

exports[`Render PrivateRoute without auth 1`] = `null`;
